packages/ra-ui-materialui/src/input/AutocompleteArrayInput.tsx

```diff
@@ -67,30 +66,45 @@ import { AutocompleteInput, AutocompleteInputProps } from './AutocompleteInput';
  */
 
 export const AutocompleteArrayInput = <
-    OptionType extends RaRecord = RaRecord,
+    RecordType extends Record<string, unknown> = never,
+    OptionType extends Record<string, unknown> = never,
```

Comment:
Why `unknown` instead of `any` here?

```suggestion
    RecordType extends Record<string, any> = never,
    OptionType extends Record<string, any> = never,
```

Besides, isn't this a **breaking change** for people using `AutocompleteArrayInput` like this today?

```tsx
<AutocompleteArrayInput<MyOptionType>>...</AutocompleteArrayInput>
```

Reply:
It's unknown because it is actually unknown so it's better semantically and a reminder that you should provide the type